### DISCUSSION TITLE
Using the GUI with existing setups/configs and systemd

### DISCUSSION THREAD

**Discussion by johnsidoli** [bpozdena/OneDriveGUI#71] [Q&A]
Hi,

I'd love to use a GUI for onedrive client, which I've used for about a year on its own very successfully.

I did try and install it, but it picked up my existing config and then overwrote it and then didn't work. I managed to get everything working again by deleting the app image and restoring my old config (was a bit scary given I depend on it for work!).

May I ask... might it not have worked because onedrive is already running as a systemd service (in my case, several times across normal onedrive and several sharepoint libraries)?

Would it need me to stop the systemd service and delete the systemd config files first?

Thanks and keep up the good work.

**Comment by bpozdena** [comment]
Hi @johnsidoli ,

Can you please share the original and modified config files?

In theory, the GUI does not make any changes to the config other than removing comments and adding missing default options. This is briefly explained in [readme](https://github.com/bpozdena/OneDriveGUI#additional-notes).

The GUI also automatically creates a config backup at `~/.config/onedrive/config_backup` before saving changes.

There should be no issues with editing the config file while your OneDrive service/process is running. You will only need to restart it to pickup config new changes.

**Comment by johnsidoli** [comment]
Unfortunately I wiped everything to get it back to previous setup - yes, it created backups which is good as I just restored them. I noticed it removed comments and defaults so I think that all worked fine. It just wouldn't sync when it loaded everything, and seemed to break, so I panicked and restored everything.

I was just wondering ... if I try again, maybe I need to remove all the systemd stuff that currently makes it all run [i.e. before running and configuring the GUI?]?

Sorry I deleted all the stuff anyway, I realise that's unhelpful for you... !
